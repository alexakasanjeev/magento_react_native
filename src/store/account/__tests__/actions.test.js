import {
  logout,
  getOrderList,
  addAccountAddress,
  resetAddressStatus,
  getCurrentCustomer,
  getOrderedProductInfo,
} from '../actions';

describe('account actions', () => {
  test('getCurrentCustomer()', () => {
    // Exercise
    const result = getCurrentCustomer();

    // Verify
    expect(result).toMatchSnapshot();
  });

  test('logout()', () => {
    // Exercise
    const result = logout();

    // Verify
    expect(result).toMatchSnapshot();
  });

  test('getOrderList()', () => {
    // Setup
    const customerId = 2;

    // Exercise
    const result = getOrderList(customerId);

    // Verify
    expect(result).toMatchSnapshot();
  });

  test('getOrderedProductInfo()', () => {
    // Setup
    const sku = 'WH02-S';

    // Exercise
    const result = getOrderedProductInfo(sku);

    // Verify
    expect(result).toMatchSnapshot();
  });

  /**
   * Note: Any data can be modified inside in customerData
   * and sent to action, and this test won't reflect it.
   */
  test('addAccountAddress()', () => {
    // Setup
    const customerId = 51;
    const customerData = {
      customer: {
        created_at: '2019-11-08 09:57:38',
        created_in: 'Default Store View',
        disable_auto_group_change: 0,
        email: 'Test3@gmail.com',
        firstname: 'Alex',
        group_id: 1,
        id: 51,
        lastname: 'Newman',
        store_id: 1,
        updated_at: '2019-11-28 10:15:49',
        website_id: 1,
        addresses: [
          {
            region: 'Arizona',
            region_id: '4',
            country_id: 'US',
            street: [
              'magical avenue'
            ],
            telephone: '123456789',
            postcode: '123456',
            city: 'dark valley',
            firstname: 'Alex',
            lastname: 'Newman',
          },
        ]
      }
    };

    // Exercise
    const result = addAccountAddress(customerId, customerData);

    // Verify
    expect(result).toMatchSnapshot();
  });

  test('resetAddressStatus()', () => {
    // Exercise
    const result = resetAddressStatus();

    // Verify
    expect(result).toMatchSnapshot();
  });
});
