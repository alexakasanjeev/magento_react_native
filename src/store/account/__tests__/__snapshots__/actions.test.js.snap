// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`account actions addAccountAddress() 1`] = `
Object {
  "payload": Object {
    "customerData": Object {
      "customer": Object {
        "addresses": Array [
          Object {
            "city": "dark valley",
            "country_id": "US",
            "firstname": "Alex",
            "lastname": "Newman",
            "postcode": "123456",
            "region": "Arizona",
            "region_id": "4",
            "street": Array [
              "magical avenue",
            ],
            "telephone": "123456789",
          },
        ],
        "created_at": "2019-11-08 09:57:38",
        "created_in": "Default Store View",
        "disable_auto_group_change": 0,
        "email": "Test3@gmail.com",
        "firstname": "Alex",
        "group_id": 1,
        "id": 51,
        "lastname": "Newman",
        "store_id": 1,
        "updated_at": "2019-11-28 10:15:49",
        "website_id": 1,
      },
    },
    "customerId": 51,
  },
  "type": "MAGENTO_ADD_ACCOUNT_ADDRESS_REQUEST",
}
`;

exports[`account actions getCurrentCustomer() 1`] = `
Object {
  "type": "MAGENTO_CURRENT_USER_REQUEST",
}
`;

exports[`account actions getOrderList() 1`] = `
Object {
  "payload": Object {
    "customerId": 2,
  },
  "type": "MAGENTO_GET_ORDERS_REQUEST",
}
`;

exports[`account actions getOrderedProductInfo() 1`] = `
Object {
  "payload": Object {
    "sku": "WH02-S",
  },
  "type": "MAGENTO_GET_ORDERED_PRODUCT_INFO_REQUEST",
}
`;

exports[`account actions logout() 1`] = `
Object {
  "type": "USER_LOGOUT",
}
`;

exports[`account actions resetAddressStatus() 1`] = `
Object {
  "type": "RESET_ADDRESS_STATUS",
}
`;
